adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -286,6 +286,13 @@ void updateCachedItemWithResult(final String resource, final String clientId, fi
                 throw new AuthenticationException(ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_URL, e.getMessage(), e);
             }
         } else if (AuthenticationConstants.OAuth2ErrorCode.INVALID_GRANT.equalsIgnoreCase(result.getErrorCode())) {
+            // make sure to set authority to the one in preferred cache
+            try {
+                mAuthority = getAuthorityUrlWithPreferredCache();
```

Comment:
Discussed in standup, going to leave as-is. If this `Exception` is thrown (it shouldn't), we'll fall-back on the passed-in Authority value and keep 🚚in